app/controllers/landing_page_controller.rb

```diff
@@ -75,68 +77,30 @@ def parse_int(int_str_or_nil)
   end
 
   def community_id(request)
-    # TODO - This will come from request.env where the to-be-implemented middleware will put the data
-    ident = request.host.split(".").first
-    if ident == "aalto"
-      501
-    elsif ident == "oin"
-      11
+    c = request.env[:current_marketplace]
+    if c
+      c.id
     end
   end
 
   def render_landing_page(cid, structure)
     locale = structure["settings"]["locale"]
+    sitename = structure["settings"]["sitename"]
```

Comment:
Ok, now we can use `values_at` again :)

Previously, the locale variable was set like this:

```
locale = structure["settings"].values_at("locale")
```

The problem with this was that `values_at` returns an array and because of that, `locale` was an array, e.g. `["en"]`. That's why I changed it to `structure["settings"]["locale"]`. But now that we have two values we can use destructuring:

```
locale, sitename = structure["settings"].values_at("locale", "sitename")
```